jgrapht-core/src/main/java/org/jgrapht/alg/cycle/JohnsonSimpleCycles.java

```diff
@@ -161,12 +166,11 @@ public void setGraph(Graph<V, E> graph)
             }
         }
         if (minSCC == null) {
-            return result;
+            return null;
         }
 
         // build a graph for the SCC found
-        Graph<V, E> resultGraph = new DefaultDirectedGraph<>(
-            new ClassBasedEdgeFactory<>((Class<? extends E>) DefaultEdge.class));
+        Graph<V, E> resultGraph = new DefaultDirectedGraph<>(graph.getEdgeFactory());
```

Comment:
After reviewing the place where we call addEdge on the result graph, I'm wondering why we don't reuse the edges from the original graph instead of creating new ones.  This would have the benefit that if the original graph was weighted (or had other edge attributes), those would be preserved on the cycles, whereas currently we will create uniform edges (e.g. all with weight 1).

Reply:
Nice catch, since the edges use the same source and target. I did the necessary changes. For now it does matter much since cycles are returned as list of vertices, however, I plan to augment the interface to also return a list of `GraphPath`s in the future.

I also added a check for "multiple" edges as the implementation does not support this.